### PR TITLE
[ISSUE #19981] testing version bump

### PR DESCRIPTION
**Pull request by maxi297**
## What
This PR addresses #19981 

Since we currently release in branches, it can lead to weird race conditions. This PR will:
* Allow the release process to occur in master
* Change the trigger from a slash command to a GitHub workflow

## How
See https://github.com/airbytehq/airbyte/issues/19981#issuecomment-1374028384

### CODE REVIEW

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:53]: Note to myself: should use input from workflow_dispatch

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:55]: Note to myself: should use input from workflow_dispatch

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:64]: Note to myself: to be confirmed with the team. If we decide this is what we want, we need to:
* choose the channel (right now, this channel id is the one from #issue-19981_test-release-notification)
* add Airbyte Tooling Bot to the channel

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:61]: Note to myself: should use input from workflow_dispatch

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:92]: TBD how we want to deal with notifications

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:100]: Uncomment when "on: push" is removed

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:134]: TBD how we want to deal with notifications

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:141]: TBD how we want to deal with notifications

**Comment by maxi297** [comment]
Should we remove the current `publish-cdk` workflow? If yes, I'll:
* remove the file
* remove the command in slash-commands.yml
* update the documentation

File: airbyte-cdk/python/.bumpversion.cfg
Comment (maxi297) [inline airbyte-cdk/python/.bumpversion.cfg:2]: Version to be downgraded once `on: push` is removed

File: airbyte-cdk/python/CHANGELOG.md
Comment (maxi297) [inline airbyte-cdk/python/CHANGELOG.md:3]: To be removed once `on: push` is removed

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [inline .github/workflows/publish-cdk-command-manually.yml:2]: To be removed once testing is completed

File: airbyte-cdk/python/setup.py
Comment (maxi297) [inline airbyte-cdk/python/setup.py:18]: To be removed once on: push is removed

File: .github/workflows/publish-cdk-command-manually.yml
Comment (brianjlai) [reply .github/workflows/publish-cdk-command-manually.yml:64]: I'm good with a designated slack channel for cdk releases. Altertnatively we could try to retain the same behavior and just post a comment at the end of the GH issue. The only difference being that now it would be after the merge

**Comment by brianjlai** [comment]
Am I understanding this correctly that our development flow will change into as follows:

1. Submit PR
2. Wait for review
3. Approval
4. Merge PR
5. Trigger workflow manually from GH (and necessary inputs)
6. Push changes to PyPi (+ dry run)
7. Octavia created a PR w/ the changes to setup.py + changelog?
8. Do we need to merge this in ourselves?

Is that about right?

File: .github/workflows/publish-cdk-command-manually.yml
Comment (sherifnada) [reply .github/workflows/publish-cdk-command-manually.yml:64]: I think we get a lot of value without notifications at first. Then we can add notifications as a follow up to this PR.

**Review by sherifnada — COMMENTED** [review]
Makes sense! I like it. 

Also, Brian's comment is a good reminder to update the docs! I think it helps in this kind of change to start with the docs/readme first, so the developer understands how the interface/workflow is changing, before talking about the implementation details.

**Comment by maxi297** [comment]
> Am I understanding this correctly that our development flow will change into as follows:
> 
> 1. Submit PR
> 2. Wait for review
> 3. Approval
> 4. Merge PR
> 5. Trigger workflow manually from GH (and necessary inputs)
> 6. Push changes to PyPi (+ dry run)
> 7. Octavia created a PR w/ the changes to setup.py + changelog?
> 8. Do we need to merge this in ourselves?
> 
> Is that about right?

@brianjlai The new flow is documented here: https://github.com/airbytehq/airbyte/issues/19981#issuecomment-1374028384 In the context of a PR, I would correct the above steps as :
1. Submit PR
2. Wait for review
3. Approval
4. Merge PR
5. Trigger workflow manually from GH (and necessary inputs). This will automatically:
    1. Octavia writes a commit message on master updating the version in setup.py and adding a changelog
    2. Push changes to PyPi

Therefore, there is no second PR. Now, we had a discussion with Sherif as to if we keep the dry-run. If we want to keep it, we could probably add a `none` to the "part-to-bump". This would change the steps as:
1. Submit PR
2. Wait for review
3. Approval
4. Merge PR
5. Trigger workflow manually from GH (dry-run=true with part-to-bump=patch|minor|major). This will automatically:
    1. Octavia writes a commit message on master updating the version in setup.py and adding a changelog
    2. Push changes to test PyPi
6. Trigger workflow manually from GH (dry-run=false with part-to-bump=none). This will automatically:
    1. Push changes to PyPi

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [reply .github/workflows/publish-cdk-command-manually.yml:64]: Posting a comment at the end of a GH issue will be challenging since this new way of releasing makes it that there is no GH issue associated with the release (at least, not that the GitHub workflow knows about)

File: .github/workflows/publish-cdk-command-manually.yml
Comment (maxi297) [reply .github/workflows/publish-cdk-command-manually.yml:100]: As mentioned by Sherif, we will remove notifications for now and add them in another PR

File: airbyte-cdk/python/README.md
Comment (girarda) [inline airbyte-cdk/python/README.md:145]: is there a reason not to do the dry run and the actual release as a single step? I don't think I ever want to only run the dry run

File: .github/workflows/publish-cdk-command-manually.yml
Comment (sherifnada) [inline .github/workflows/publish-cdk-command-manually.yml:13]: nit: i'd rename to `release type` or something since the release itself is either major/minor/bump

**Review by sherifnada — APPROVED** [review]
LGTM assuming suggested changes + code cleanup

File: airbyte-cdk/python/README.md
Comment (brianjlai) [reply airbyte-cdk/python/README.md]: I think it was just an extra sanity check for what we were trying to pushwas valid since the push to PyPi seems pretty permanent irreversible. I'm not super attached, but I would be fine leaving it just because a few extra minutes on what is already an automated process feels acceptable.

**Review by brianjlai — APPROVED** [review]

File: airbyte-cdk/python/README.md
Comment (maxi297) [reply airbyte-cdk/python/README.md]: Agreed! I'll keep it but we can reassess if this causes any kind of issue later on (like "this is flaky in the new process" or "we are changing things in this pipeline and maintaining this is costly")

**Pull request merged**
